pkg/icingaredis/telemetry/heartbeat.go

```diff
@@ -0,0 +1,135 @@
+package telemetry
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/go-redis/redis/v8"
+	"github.com/icinga/icingadb/internal"
+	"github.com/icinga/icingadb/pkg/com"
+	"github.com/icinga/icingadb/pkg/icingaredis"
+	"github.com/icinga/icingadb/pkg/logging"
+	"github.com/icinga/icingadb/pkg/periodic"
+	"github.com/icinga/icingadb/pkg/utils"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"runtime/metrics"
+	"strconv"
+	"sync/atomic"
+	"time"
+)
+
+// ha represents icingadb.HA to avoid import cycles.
+type ha interface {
+	State() (weResponsibleMilli int64, weResponsible, otherResponsible bool)
+}
+
+// jsonMetricValue allows JSON-encoding a metrics.Value.
+type jsonMetricValue struct {
+	v *metrics.Value
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (jmv jsonMetricValue) MarshalJSON() ([]byte, error) {
+	switch kind := jmv.v.Kind(); kind {
+	case metrics.KindUint64:
+		return json.Marshal(jmv.v.Uint64())
+	case metrics.KindFloat64:
+		return json.Marshal(jmv.v.Float64())
+	default:
+		return nil, errors.Errorf("can't JSON-encode Go metric value of kind %d", int(kind))
+	}
+}
+
+type SuccessfulSync struct {
+	WhenMilli int64
+	TookMilli int64
```

Comment:
`WhenMilli` could be named better so that it's clear whether it's the start or finish timestamp.

Also, I'd prefer `Duration` over `Took` here as well.